File: kinotic-js/workspace/packages/os-api/src/api/model/workload/PortMapping.ts
import type { PortProtocol } from '@/api/model/workload/PortProtocol'

/**
 * A guest port exposed on the host for a workload's VM.
 */
export interface PortMapping {

    /** Host port to expose. When omitted, the provider assigns one. */
    hostPort?: number

    /** Guest port to map to the host. */
    guestPort: number

    /** Transport protocol. When omitted, the provider's default applies. */
    protocol?: PortProtocol

    /** Host interface to bind on. When omitted, the provider's default applies. */
    hostIp?: string
}

File: kinotic-domain/src/main/java/org/kinotic/domain/api/model/workload/PortMapping.java
package org.kinotic.domain.api.model.workload;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * A guest port exposed on the host for a {@link Workload}'s VM.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class PortMapping {

    /**
     * Host port to expose. When null, the provider assigns one.
     */
    private Integer hostPort;

    /**
     * Guest port to map to the host.
     */
    private int guestPort;

    /**
     * Transport protocol. When null, the provider's default applies.
     */
    private PortProtocol protocol;

    /**
     * Host interface to bind on. When null, the provider's default applies.
     */
    private String hostIp;
}
